sorting/stooge_sort.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file
+ * @brief [Stooge sort implementation](https://en.wikipedia.org/wiki/Stooge_sort)
+ * in C++
+ * @details
+ * Stooge sort is a recursive sorting algorithm.
+ * It divides the array into 3 parts and proceeds to:
+ *	- sort first two thirds of the array
+ *	- sort last two thirds of the array
+ *  - sort first two thirds of the array
+ * It's time complexity is O(n^(log3/log1.5)), which is about O(n^2.7), 
+ * which makes it to be not the most efficient sorting algorithm 
+ * on the street on average. Space complexity is O(1).
+ */
+
+#include <vector>
+#include <iostream>
+
+void stoogeSort(std::vector<int> *L, int i, int j) {
+	if (i < j) {
+		if ((*L)[i] > (*L)[j]) {	
+			std::swap((*L)[i], (*L)[j]);
+		}
+		if (j - i > 1) {
+			int third = (j - i + 1) / 3;
+			stoogeSort(L, i, j - third);
+			stoogeSort(L, i + third, j);
+			stoogeSort(L, i, j - third);
+		}
+	}
+	return;
+}
+
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃

Reply:
Just added documentation and 3 test cases, thanks!